src/main/java/com/sivalabs/bookstore/catalog/domain/ProductService.java

```diff
@@ -53,4 +53,14 @@ private TextCriteria textCriteriaMatching(String searchCriteria) {
     private Pageable pageableSortOf(int page) {
         return PageRequest.of(page, PAGE_SIZE, Sort.Direction.ASC, "name");
     }
+
+    public Optional<Product> deleteProduct(String code) {
```

Comment:
This logic can be simplified.

```
Product product = productRepository.findByCode(code).orElseThrow(()-> new ProductNotFoundException(code));
product.setDeleted(true);
productRepository.save(product);
```

I think we don't need to return the product from delete method.